public/app/features/scenes/dashboard/utils.ts

```diff
@@ -0,0 +1,26 @@
+import { SceneDeactivationHandler, SceneObject } from '@grafana/scenes';
+
+export function getVizPanelKeyForPanelId(panelId: number) {
+  return `panel-${panelId}`;
```

Comment:
This makes me slightly nervous, but not enough to think it is a really bad idea.  In https://github.com/grafana/grafana-iot-twinmaker-app we define a complex application by dynamically replacing the layout with the layout from a set of dashbaords that depend on conditions (template variables and basic evaluations). In this scenario, we may end up with multiple VizPanels that have the same key 🤔

Reply:
@ryantxu but panel ids should be unique, right?